iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -544,10 +556,12 @@ class string
     friend class string;
 
     template <typename T1, typename T2>
-    friend typename std::enable_if<(is_char_array<T1>::value || is_cxx_string<T1>::value)
-                                       && (is_char_array<T2>::value || is_cxx_string<T2>::value),
-                                   string<internal::GetCapa<T1>::capa + internal::GetCapa<T2>::capa>>::type
-    concatenate(const T1& t1, const T2& t2) noexcept;
+    friend
+        typename std::enable_if<(is_char_array<T1>::value || is_cxx_string<T1>::value || std::is_same<T1, char>::value)
+                                    && (is_char_array<T2>::value || is_cxx_string<T2>::value
+                                        || std::is_same<T2, char>::value),
+                                string<internal::GetCapa<T1>::capa + internal::GetCapa<T2>::capa>>::type
```

Comment:
Could you create for this one please a constexpr to improve readability. Something like:
```cxx
template<typename T1, typename T2>
constexpr uint64_t CAPACITY_SUM = internal::GetCapa<T1>::capa + internal::GetCapa<T2>::capa;

// could be used like
cxx::string<CAPACITY_SUM<StringType1, StringType2>> someString;
```